### ISSUE TITLE
User Profile Stats

### ISSUE THREAD

**Issue by svvoii** [svvoii/ft_transcendence#65]
- User profile display stats, such as wins and losses.

**Comment by svvoii** [comment]
Working on this..

**Comment by Drewski6** [comment]
Ok sounds good. When you get that model working and have an api endpoint I have send a request to, let me know and I'll add it to the front

**Comment by svvoii** [comment]
So, along with #64, I have added an API endpoint to retrieve the user game stats from the UserGameStat model. It’s in the `a_user/views.py` line 301, `api_user_game_stats_view`.
Also, its paired fetch function can be found in the new file : `../js/user/UserAPI.js` - `getUserGameStats` on line 78.

I guess the very next step is to test those endpoints once the functionality is available on the front-end. Thanks.

**Comment by Drewski6** [comment]
great! I'm taking a look into this now. I think I'll make my changes to the front directly on your user_game_stats branch if that's ok

**Comment by Drewski6** [comment]
the subject says "Each user has a Match History including 1v1 games, dates, and relevant
details, accessible to logged-in users.", I'm reading this as "each player should be able to see the match history of every other player and themselves".
given this, I'm going to make the match history of every player available to anyone who is logged in.

**Comment by Drewski6** [comment]
"Each user has a Match History including 1v1 games, dates, and relevant
details, accessible to logged-in users."
We have the data from 1v1 games, we still need the dates. I think we should return an object with each match the player has played, the score, and the time/date it occured along with the information already provide. 

I'd like to see something like this maybe:
```
{ 
   stats_str: "test2's game stats", 
   total_games_played: 0, 
   total_wins: 0, 
   total_losses: 0,
   match_data: {
      match1: {
         player1_score: 3,
         player2_score: 10,
         date_time: 1234123412341234,
      },
      match2: ... etc   
   }
}
```
Or something close to this,

I think this will allow us to give fulfill the requirements detailed in the subject.

**Comment by svvoii** [comment]
Yes, that means expanding the data returned by the endpoint. That is what I am going to do. 
Thanks.

**Comment by svvoii** [comment]
@Drewski6 
I added another endpoint `user/match_history/<str:username>/` which returns… what the name is :D (you will find the view function on the line 332 of `a_user/views.py`).

There is still another endpoint for game stats.

I suggest we make two buttons, one for `Game Stats`, another is for `Match History`, or another button to click from the stats view to see the match history. What do you think ?

I didn't test the endpoint.

**Comment by svvoii** [comment]
Fixed with PR #85
